### PR TITLE
[SWUI-17][FEAT] - Create ButtonLink component

### PR DESCRIPTION
**Pull request by ElRodrigote**
## Fixes: [SWUI-17](https://linear.app/swaprhq/issue/SWUI-17/create-buttonlink)

# Description
* Move the `Button` component and its story to its folder
* Create the `ButtonLink` component and its story

# Visual Evidence
image

### CODE REVIEW

File: src/components/Button/Button.tsx [hunk 1]
```diff
@@ -0,0 +1,48 @@
+import React, { ButtonHTMLAttributes } from "react";
+
+import { twMerge } from "tailwind-merge";
+
+import { buttonStyles } from "./utils";
```
Comment (Diogomartf) [inline src/components/Button/Button.tsx:5]: it doesn't make sense the styles coming from "utils"
if we want another file for styles it should be called `styles`.

But I'm okay with declaring the styles here and import them on `ButtonLink`

**Comment by Diogomartf** [comment]
Related to styles, they seem to be broken.

image

`ButtonLink` is just a `Button` in terms of styles, the only difference is one is a link element and the other is a button element.

File: src/components/ButtonLink/ButtonLink.tsx [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+
+export interface ButtonLinkProps<T extends React.ElementType>
+  extends ButtonProps {
+  as?: T;
+  children?: React.ReactNode;
+  href: string;
+  iconLeft?: IconName;
+  iconRight?: IconName;
+  target?: string;
+}
```
Comment (berteotti) [inline src/components/ButtonLink/ButtonLink.tsx:16]: Let's not add new props to `ButtonLink`, and let's not extend `ButtonProps`. `ButtonLinkProps` should only be this: 

```suggestion
export type ButtonLinkProps<T extends React.ElementType> = React.PropsWithChildren<
   React.ComponentPropsWithoutRef<T> & {
      as?: T
   }
```

File: src/components/ButtonLink/ButtonLink.tsx [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+      id={id}
+      tabIndex={0}
+      target={target}
+      {...props}
+    >
+      {iconLeft && <Icon size={iconSize} name={iconLeft} />}
+      {children && <div>{children}</div>}
+      {iconRight && <Icon size={iconSize} name={iconRight} />}
+    </Component>
+  );
```
Comment (berteotti) [inline src/components/ButtonLink/ButtonLink.tsx:57]: Let's remove all unnecessary props. This should be enough for `ButtonLink`.

```suggestion
}: ButtonLinkProps<T>) {
  const Component = as || "a";

  return (
    <Component
      className={twMerge(
        buttonStyles({
          className,
          size,
          variant,
          width,
        }),
        className
      )}
      tabIndex={0}
      {...props}
    />
  );
```

File: src/components/ButtonLink/ButtonLink.tsx [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+  className,
+  href,
+  iconLeft,
+  iconRight,
+  id,
+  size,
+  target,
+  variant,
+  width,
+  ...props
```
Comment (berteotti) [inline src/components/ButtonLink/ButtonLink.tsx:30]: Same here, only a few props needed (`as` and the ones related to styling).
Other props can be passed through `...props`.

```suggestion
  as,
  className,
  size,
  variant,
  width,
  ...props
```

File: src/components/ButtonLink/ButtonLink.tsx (hunk 2 above)
Comment (ElRodrigote) [reply src/components/ButtonLink/ButtonLink.tsx]: Implementing it!

**Review by berteotti — APPROVED** [review]
Looking good, good job Rorry!

**Pull request merged**